PGM/src/main/java/tc/oc/pgm/match/MatchLoader.java

```diff
@@ -165,7 +165,7 @@ private Match cycleToUnsafe(@Nullable Match oldMatch, PGMMap newMap) throws Thro
      * @param map Map to load.
      */
     private Match loadMatch(PGMMap map) throws Throwable {
-        return map.getContext().asCurrentScope(() -> {
+        return map.getContext().orElseThrow(MapNotFoundException::new).asCurrentScope(() -> {
```

Comment:
The map context is reloaded on MatchManager#cycleTo before starting a cycle, if the context has been GCed before reaching this point, there's a serious memory shortage...